nomad/state/schema.go

```diff
@@ -619,3 +620,30 @@ func schedulerConfigTableSchema() *memdb.TableSchema {
 		},
 	}
 }
+
+func csiVolumeTableSchema() *memdb.TableSchema {
```

Comment:
We may also want an index on the Namespace? and or store them by namespace? This would allow name collisions across namespaces which would simplify deploying jobs to staging/prod with minimal config changes, and would also maybe make some lookups more efficient?

Reply:
I made this change, and I now have followup questions:

1. Should listing by `Driver` default to namespace specific too?
2. via UI converstions, it seems like we're all saying `Provider` now. Is that better language, because if so I'll rename `Driver`

Reply:
`Driver` is the name that is used by Kubernetes for CSI Plugins, and so I'd rather us keep that to avoid confusion. `Storage Provider` in CSI terms refers to an external entity, rather than a plugin itself

Reply:
`Driver` should probably be namespace specific, I think.

Reply:
The real question is, should we let a single driver deployment provide volumes to multiple namespaces - my gut is no because it seems like permissioning would get confusing?

Reply:
Ok, so does that mean avoiding namespaces on the volumes too? it seems like it would. I undid all the namespacing in a single commit, so we can leave it out or in pretty easily.